AdyenCard/Validators/CardNumberValidator.swift

```diff
@@ -38,12 +40,30 @@ public final class CardNumberValidator: Validator {
         self.panLength = panLength
     }
     
-    public func isValid(_ value: String) -> Bool {
-        guard isEnteredBrandSupported else { return false }
-        let minimumValidCardLength = 12
-        let isValid = value.count >= minimumValidCardLength && (!isLuhnCheckEnabled || luhnCheck(value))
+    public func validate(_ value: String) -> ValidationStatus {
+        // order of checks are important to return the correct error.
+        
+        if !isEnteredBrandSupported {
```

Comment:
Just a thought - maybe something for a later point - but reading this feels a bit odd that we're passing a `isEnteredBrandSupported` to the CardNumberValidator which basically overrides all other validation rules.